src/renderers/dom/shared/DOMProperty.js

```diff
@@ -227,26 +220,150 @@ var DOMProperty = {
     }
   },
 
-  getPropertyInfo(name) {
-    return DOMProperty.properties.hasOwnProperty(name)
-      ? DOMProperty.properties[name]
-      : null;
+  getAttributeName(propName) {
+    return attributeNames.hasOwnProperty(propName)
+      ? attributeNames[propName]
+      : propName;
+  },
+
+  getAttributeNamespace(propName) {
+    switch (propName) {
+      case 'xmlBase':
+      case 'xmlLang':
+      case 'xmlSpace':
+        return 'http://www.w3.org/XML/1998/namespace';
+      case 'xlinkActuate':
+      case 'xlinkArcrole':
+      case 'xlinkHref':
+      case 'xlinkRole':
+      case 'xlinkShow':
+      case 'xlinkTitle':
+      case 'xlinkType':
+        return 'http://www.w3.org/1999/xlink';
+      default:
+        return null;
+    }
+  },
+
+  // If non-null, used instead of the property or `setAttribute()`.
+  getMutationMethod(propName) {
+    switch (propName) {
+      case 'value':
+        return setDOMValueAttribute;
+      default:
+        return null;
+    }
+  },
+
+  getExpectedValueType(propName) {
+    switch (propName) {
+      // Numeric properties.
+      // TODO: add an explanation.
+      case 'rowSpan':
+      case 'start':
+        return 'number';
+      // Positive numeric properties.
+      // TODO: add an explanation.
+      case 'cols':
+      case 'rows':
+      case 'size':
+      case 'span':
+        return 'positiveNumber';
+      // Booleans.
+      // The property should be removed when set to a falsey value.
+      // case 'autoFocus': (Intentionally handled elsewhere.)
+      case 'allowFullScreen':
+      case 'async': // Specifies target context for links with `preload` type.
+      case 'autoPlay':
+      case 'capture':
+      case 'checked':
+      case 'controls':
+      case 'default':
+      case 'defer':
+      case 'disabled':
+      case 'formNoValidate':
+      case 'hidden':
+      case 'itemScope': // For Microdata. http://schema.org/docs/gs.html
+      case 'loop':
+      case 'multiple':
+      case 'muted':
+      case 'noValidate':
+      case 'open':
+      case 'playsInline':
+      case 'readOnly':
+      case 'required':
+      case 'reversed':
+      case 'scoped':
+      case 'seamless':
+      case 'selected':
+        return 'boolean';
+      // "String" booleans (TODO: explain the difference).
+      case 'allowTransparency':
+      case 'contentEditable':
+      case 'draggable':
+      case 'spellCheck':
+      case 'value':
+      case 'autoReverse':
+      case 'externalResourcesRequired':
+      case 'preserveAlpha':
+        return 'stringBoolean';
+      // Booleans or strings.
+      // The property can be used as a flag as well as with a value.
+      // Removed when strictly equal to false; present without a value when
+      // strictly equal to true; present with a value otherwise.
+      case 'download':
+        return 'overloadedBoolean';
```

Comment:
Yes, flowifying and turning into something simper is in the plan. (Although it's already smaller than the original.)

Reply:
AFAIK v8 used to/does deop switches without string literals as the case, I couldn't find anything recent that gave clarity on if that was still true, it may not matter here either. (I mention it because  the return here is used in a switch elsewhere yeah?)